### ISSUE TITLE
RemoteCluster resource

### ISSUE THREAD

**Issue by klizhentas** [gravitational/teleport#1526]
## Description

Whenever TrustedCluster connects to the main cluster, main cluster has the connection and trust established, but there is no way to delete/update the trust from the main server perspective.

### RemoteCluster

Resource `RemoteCluster` is added for visibility and control on the main cluster side whenever remote trusted cluster connects to it.

```yaml
version: v3
kind: remote_cluster
metadata:
   name: <remote cluster name>
# status contains fields that are populated dynamically and are read only,
# are here for diangostic purposes
status:
  connection: online
```

Remote cluster name is the same name as set by the trusted cluster in the config.

### Listing remote clusters

Listing is for diangostic/inspection purposes

```
#
# rc stands for remote clusters
tctl get rc
# same as rc
tctl get remoteclusters
# get cluster by name
tctl get rc/cluster-name
```

### Deleting remote cluster

Deleting remote cluster will lead to the trust deleted on the main cluster side and connection forcibly dropped.

```
#
# rc stands for remote clusters

# rm cluster by name
tctl rm rc/cluster-name
```

Trusted cluster can re-initiate the trust by using the existing trusted cluster flow assuming the token is  valid.

Thoughts: it will be nice, but not required for the first release to gracefully handle the situation on the existing remote cluster side, assuming the remote trusted cluster is connected, it will be advised for remote cluster to remove the trust as well - the trusted cluster resource on the remote cluster side will stay, but active reverse tunnel connection and certificate authorities should be removed. Need to think about this edge case a bit more

**Comment by kontsevoy** [comment]
Is this for our "AWS Milestone"? Thoughts?

**Comment by klizhentas** [comment]
@kontsevoy this is implemented in master, will go into 2.5.0 "AWS" release
